source/scripts/InjectedScript/utils.ts

```diff
@@ -0,0 +1,17 @@
+enum StargazerChain {
+  CONSTELLATION = 'constellation',
+  ETHEREUM = 'ethereum',
+}
+
+const readOnlyProxy = <T extends object>(object: T): T => {
+  return new Proxy(object, {
+    set: () => false,
+    defineProperty: () => false,
+    deleteProperty: () => false,
+    setPrototypeOf: () => false,
+  });
```

Comment:
I think you actually have this backwards. If I set a `set` method on `object` it will override the `set` defined in your 2nd parameter object. 

Would Object.freeze handle this better?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

Reply:
Depending on the implementation it does or does not work as intended... If no other reference to the source object is given it should work as intended...